packages/subcommands/src/events/command-handler/PreSubcommandRun.ts

```diff
@@ -0,0 +1,66 @@
+import { CommandStore, Event, Events, isErr, PreCommandRunPayload, UserError } from '@sapphire/framework';
+import type { PieceContext } from '@sapphire/pieces';
+import { SubCommandsEvents } from '../../lib/utils/SubCommandsHandler';
+
+export interface PreSubCommandPayload extends PreCommandRunPayload {}
+
+export class PluginEvent extends Event<SubCommandsEvents.PreSubCommandRun> {
+	public constructor(context: PieceContext) {
+		super(context, { event: SubCommandsEvents.PreSubCommandRun });
+	}
+
+	public async run({ message, command: baseCommand, parameters, context }: PreSubCommandPayload) {
+		try {
+			const { command, args, method, parentName } = await CommandStore.injectedContext.subCommandsHandler.resolveSubCommand({
+				command: baseCommand,
+				message,
+				parameters
+			});
+
+			if (command || (method && baseCommand && Reflect.has(baseCommand, method))) {
+				const resolvedCommand = command ?? baseCommand;
+
+				if (!resolvedCommand.enabled) {
+					message.client.emit(SubCommandsEvents.SubCommandDenied, new UserError('SubCommandDisabled', 'This sub command is disabled.'), {
+						message,
+						command: resolvedCommand,
+						parameters,
+						context: { ...context, parentName }
+					});
+
+					return;
+				}
+
+				const result = await resolvedCommand.preconditions.run(message, resolvedCommand);
+
+				if (isErr(result)) {
+					message.client.emit(SubCommandsEvents.SubCommandDenied, result.error, {
+						message,
+						command: resolvedCommand,
+						parameters,
+						context: { ...context, parentName }
+					});
+				} else {
+					message.client.emit(SubCommandsEvents.SubCommandAccepted, {
+						message,
+						command: resolvedCommand,
+						method,
+						args,
+						parameters,
+						context: { ...context, parentName }
+					});
+				}
+
+				return;
+			}
+		} catch (error) {
+			message.client.emit(SubCommandsEvents.SubCommandError, error, { piece: baseCommand, message });
+			return;
+		}
```

Comment:
This is VERY bad practice. And, I'm also 99% sure @kyranet will not approve of such a huge try-catch either. Where possible try to restrict try-catch blocks only to where necessary.

Reply:
@Nytelife26 I'm not sure about this. In current block is handles code line 14 and we need to use these variables in big if block, but variables are defined inside try-catch block and are not available cause closure. "A huge" try-catch block is just an "illusion" I think

Reply:
Split this into separate methods so the scoping of the variables don't get messy, that's how.